repos/keg-core/core/scripts/js/getKegEnvs.js

```diff
@@ -0,0 +1,88 @@
+const { get, isStr } = require('@keg-hub/jsutils')
+const getAppConfig = require('@keg-hub/tap-resolver/src/resolvers/getAppConfig')
+const fs = require('fs')
+const path = require('path')
+
+/**
+ * @param {string} tapPath - path to the current tap's root
+ * @param {string} corePath - path to keg-core's root
+ * @return {Object} - object containing tap and core's app configs and package json objects
+ *  - used to resolve replacement values for envs of form "<tap|core>.<package|config>.<some_property>"
+ */
+const buildReplaceContext = (tapPath, corePath) => {
+  const buildContext = rootPath => ({
+    package: rootPath
+      ? require(path.join(rootPath, 'package.json') )
+      : {},
+    config: rootPath
+      ? getAppConfig(rootPath, false, false)
+      : {}
+  })
+
+  return {
+    core: buildContext(corePath),
+    tap: buildContext(tapPath),
+  }
+}
+
+/**
+ * 
+ * @param {Object} replaceContext - context object produced by `buildReplaceContext`
+ * @return {Object} - object of env entries pulled from tap config and core config, 
+ * dynamically replacing any values of form `<tap|core>.<package|config>.<some_property>`
+ */
+const buildEnvs = replaceContext => {
+  const tapEnvs = get(replaceContext, 'tap.config.keg.envs')
+  const coreEnvs = get(replaceContext, 'core.config.keg.envs')
+
+  const envs = {
+    ...coreEnvs,
+    ...tapEnvs
+  }
+
+  const getFallback = value => 
```

Comment:
Not a big deal, but I typically don't create inline functions, unless I have to.
It's easier to maintain long term, and is consistent with single purpose functions methodology. Which I try to practice, but don't always succeed.

Plus if you ever need the method some where else, it's easier to make it reusaeable

Reply:
I usually don't, but I think this function is so specific to its parent function that it makes most sense to keep it in this scope.

However, thinking about that some more, I realized I don't need it to be a function at all. I can just do that work inside the reduce loop, with less overall code.